src/test/regress/pg_regress.c

```diff
@@ -940,7 +940,7 @@ convert_sourcefiles_in(const char *source_subdir, const char *dest_dir, const ch
 static void
 convert_sourcefiles(void)
 {
-	content_zero_hostname = get_host_name(-1, 'p');
+	content_master_hostname = get_host_name(-1, 'p');
```

Comment:
The upstream codes are:
> content_zero_hostname = get_host_name(0, 'p');  

Not sure why it doesn't show the 0 -> -1 differences.

Better not to change them `content_zero_hostname`,  if we are singlenode, get_host_name(-1, 'p'), else doesn't need to change.

Reply:
you means we should probe whether the server is in singlenode mode and then `get_host_name(-1, 'p')` and keep the rest of code the same?

Reply:
> you means we should probe whether the server is in singlenode mode and then get_host_name(-1, 'p') and keep the rest of code the same?

yeah

Reply:
`initialize_environment` which calls `convert_sourcefiles` happens very early in pg_ctl's initialization, during which the database even havn't started yet. I have to take `convert_sourcefiles` and call it just before running the actual tests. dunno if it's okay.

see 27757d781fff1f324f59d7237b0f3e60f529374e